e2e/__tests__/hasteMapMockChanged.test.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import JestHasteMap from 'jest-haste-map';
+import {cleanup, writeFiles} from '../Utils';
+
+// Directory must be here for Watchman to be enabled.
+const DIR = path.resolve(__dirname, 'haste_map_mock_changed');
+
+beforeEach(() => cleanup(DIR));
+afterEach(() => cleanup(DIR));
+
+test('should not warn when a mock file changes', async () => {
```

Comment:
I like `jest-haste-map` tests as e2e tests. The tests in `packages/jest-haste-map` are really hard on testing internals - this way it's easier to test it behaves properly in a way observable from the outside